src/stream_base.h

```diff
@@ -249,6 +253,8 @@ class StreamResource {
   uint64_t bytes_read_ = 0;
   uint64_t bytes_written_ = 0;
 
+  bool do_try_write_enabled_ = true;
```

Comment:
Can we make this into a bitset (`int` + flags as an enum)? It would be cool if we could merge at least something like `kHasTryWrite` (that we would unset here) and `kHasWantsWrite` (currently implemented as a virtual method).

That would allow us to skip the `DoTryWrite()` call for those streams which don’t implement it anyway a bit faster…

Reply:
Sure, that seems like a good idea!